### DISCUSSION TITLE
What is the upper limit of the number of N-wires?

### DISCUSSION THREAD

**Discussion by hihihowareyou** [PlusToolkit/PlusLib#1200] [Q&A] [answered]
What is the upper limit of the number of N-wires?

I am trying to calibrate my probe with 8 group of wires. 

So many groups are used becasue I am try to cover the image space by the wires so I can reduce the relative movement between the transducer and the phantom.

However, they are not recognized, though image quality is still OK. 

I tried to change the config.xml, the detection alogrithm can only work when the upper 3 groups are left in it. Changing groups or add group number will make it fail again. 

Any advise on large number of N-wires?

![image](https://github.com/user-attachments/assets/70280f8c-0fd3-4089-82c5-dc1ef83cb30e)
![image](https://github.com/user-attachments/assets/e47243d0-93fa-4672-914c-f4986aa5b717)
![image](https://github.com/user-attachments/assets/8cf3827e-9757-4013-b530-cb7fd191e7d2)

**Comment by lassoan** [comment]
All you need to do is to specify all the N-wires in your device set configuration file. The example configuration files contain 3 `<Pattern Type="NWire">` elements. Your configuration will contain 8.

**Reply by hihihowareyou** [reply]
Hi Lassoan, If I put 8 patterns in the configuration like below, the wire detection can't work. 

    `  <Pattern Type="NWire">
        <Wire Name="  1:G10_G10" EndPointFront=" 29.8 0 45  " EndPointBack="  29.8 70 45  "/>
        <Wire Name="  2:F10_L10" EndPointFront=" 24.8 0 45  " EndPointBack="  54.8 70 45  "/>
        <Wire Name="  3:K10_K10" EndPointFront=" 49.8 0 45  " EndPointBack="  49.8 70 45  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  4:G9_G9" EndPointFront=" 29.8 0 40  " EndPointBack="  29.8 70 40  "/>
        <Wire Name="  5:L9_F9" EndPointFront=" 54.8 0 40  " EndPointBack="  24.8 70 40  "/>
        <Wire Name="  6:K9_K9" EndPointFront=" 49.8 0 40  " EndPointBack="  49.8 70 40  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  7:I8_I8" EndPointFront=" 39.8 0 35  " EndPointBack="  39.8 70 35  "/>
        <Wire Name="  8:H8_N8" EndPointFront=" 34.8 0 35  " EndPointBack="  64.8 70 35  "/>
        <Wire Name="  9:M8_M8" EndPointFront=" 59.8 0 35  " EndPointBack="  59.8 70 35  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  10:J7_J7" EndPointFront=" 44.8 0 30  " EndPointBack="  44.8 70 30  "/>
        <Wire Name="  11:O7_I7" EndPointFront=" 69.8 0 30  " EndPointBack="  39.8 70 30  "/>
        <Wire Name="  12:N7_N7" EndPointFront=" 64.8 0 30  " EndPointBack="  64.8 70 30  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  13:E6_E6" EndPointFront=" 19.8 0 25  " EndPointBack="  19.8 70 25  "/>
        <Wire Name="  14:D6_J6" EndPointFront=" 14.8 0 25  " EndPointBack="  44.8 70 25  "/>
        <Wire Name="  15:I6_I6" EndPointFront=" 39.8 0 25  " EndPointBack="  39.8 70 25  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  16:D5_D5" EndPointFront=" 14.8 0 20  " EndPointBack="  14.8 70 20  "/>
        <Wire Name="  17:I5_C5" EndPointFront=" 39.8 0 20  " EndPointBack="  9.8 70 20  "/>
        <Wire Name="  18:H5_H5" EndPointFront=" 34.8 0 20  " EndPointBack="  34.8 70 20  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  19:G4_G4" EndPointFront=" 29.8 0 15  " EndPointBack="  29.8 70 15  "/>
        <Wire Name="  20:L4_F4" EndPointFront=" 54.8 0 15  " EndPointBack="  24.8 70 15  "/>
        <Wire Name="  21:K4_K4" EndPointFront=" 49.8 0 15  " EndPointBack="  49.8 70 15  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  22:G3_G3" EndPointFront=" 29.8 0 10  " EndPointBack="  29.8 70 10  "/>
        <Wire Name="  23:F3_L3" EndPointFront=" 24.8 0 10  " EndPointBack="  54.8 70 10  "/>
        <Wire Name="  24:K3_K3" EndPointFront=" 49.8 0 10  " EndPointBack="  49.8 70 10  "/>
      </Pattern>`

If I change the configuration as below, it works.


      <Pattern Type="NWire">
        <Wire Name="  1:G10_G10" EndPointFront=" 29.8 0 45  " EndPointBack="  29.8 70 45  "/>
        <Wire Name="  2:F10_L10" EndPointFront=" 24.8 0 45  " EndPointBack="  54.8 70 45  "/>
        <Wire Name="  3:K10_K10" EndPointFront=" 49.8 0 45  " EndPointBack="  49.8 70 45  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  4:G9_G9" EndPointFront=" 29.8 0 40  " EndPointBack="  29.8 70 40  "/>
        <Wire Name="  5:L9_F9" EndPointFront=" 54.8 0 40  " EndPointBack="  24.8 70 40  "/>
        <Wire Name="  6:K9_K9" EndPointFront=" 49.8 0 40  " EndPointBack="  49.8 70 40  "/>
      </Pattern>
      <Pattern Type="NWire">
        <Wire Name="  7:I8_I8" EndPointFront=" 39.8 0 35  " EndPointBack="  39.8 70 35  "/>
        <Wire Name="  8:H8_N8" EndPointFront=" 34.8 0 35  " EndPointBack="  64.8 70 35  "/>
        <Wire Name="  9:M8_M8" EndPointFront=" 59.8 0 35  " EndPointBack="  59.8 70 35  "/>
      </Pattern>


Please note that I am using a customized phantom. I had succeed in one trail with 5 N-wire patterns, which are all aligned in horizontal direction. I think , one possible cause of the detection failure is the horizontal offset of the wires, how do you think?

**Reply by lassoan** [reply, marked as answer]
Probably you need to relax the tolerances

**Reply by hihihowareyou** [reply]
I had tried that too, still not work. I am wondering, whether the horizontal offset affect the line detection? 

![image](https://github.com/user-attachments/assets/3cd7a6d8-18fc-4825-8a9e-3da6522ef884)

**Reply by hihihowareyou** [reply]
I had another setup, 5 patterns are positioned vertically, it works. 
But when I introduce horizontal offset, it failed.  You can check the plans as below. 
![image](https://github.com/user-attachments/assets/ff36f040-80b0-44b0-8e7d-a37f6e8712d0)

**Reply by lassoan** [reply]
Probably the xml file does not match the actual geometry of the n-wires.

**Reply by hihihowareyou** [reply]
I am quite confident about the config. 
You can check the screen shot below: the rows are quite accurate, but detection was not working. Even if I reduced the quantity of patterns to 4, (first 4 rows, quite accurate as you can see), detection still can't work. 
![image](https://github.com/user-attachments/assets/aec8a39a-fd42-4686-ab31-f448deccb34c)

**Reply by hihihowareyou** [reply]
> Probably you need to relax the tolerances

Increasing the MaxLineShiftMm from 10 to 35, problem solved. Thanks for the advice.
